### PR TITLE
Submission of QHYCCD Focuser Driver for INDI

### PR DESCRIPTION
**Pull request by NYiwtisolatedall**
Dear INDI Team,

I am an employee of QHYCCD.The content submitted this time is the driver of our company's new focusing device.

For any necessary confirmation or inquiries, please reach me at cjq@qhyccd.com.

Thank you for your consideration.

Best regards,
QHYCCD Team

### CODE REVIEW

**Comment by knro** [comment]
Thank you!

1. Why is OpenCV required? it's a huge dependency.
2. Please use [JSON C++ library](https://github.com/indilib/indi/blob/master/libs/json.h) already included with INDI.

**Comment by knro** [comment]
Hello @NYiwtisolatedall , any update on this PR? Please let me know if you need any help.

**Comment by NYiwtisolatedall** [comment]
> Hello @NYiwtisolatedall , any update on this PR? Please let me know if you need any help.

Hello @knro , I apologize for the delay in responding. I've made the recommended changes to the drive as per your last advice. Currently, I'm on a business trip. Once I return, I'll conduct testing before submitting. Is that acceptable? Thank you for your attention.

**Comment by knro** [comment]
Sure, take all the time you need.

**Comment by knro** [comment]
Any update?

File: indi-qhy/qhy_focuser.h [hunk 1]
```diff
@@ -91,6 +105,11 @@ class QFocuser : public INDI::Focuser
 
   INumber FocusSpeedN[1];
   INumberVectorProperty FocusSpeedNP;
+
+  // Reverse Focuser
+  ISwitch FocusReverseS[2];
+  ISwitchVectorProperty FocusReverseSP;
+
```
Comment (knro) [inline indi-qhy/qhy_focuser.h:112]: No need to define those, these are already defined in parent class.

File: indi-qhy/qhy_focuser.h [hunk 2]
```diff
@@ -29,47 +33,57 @@ class QFocuser : public INDI::Focuser
...
   INumber TemperatureN[1];
   INumberVectorProperty TemperatureNP;
+  INumber Voltage[1];
+  INumberVectorProperty VoltageVP;
+
+  INumber FOCUSVersion[1];
+  INumberVectorProperty FOCUSVersionVP;
+
+  INumber BOARDVersion[1];
+  INumberVectorProperty BOARDVersionVP;
```
Comment (knro) [inline indi-qhy/qhy_focuser.h:86]: Please use INDI::PropertyNumber instead as INumber/INumberVectorProperty is deprecated for new drivers.

File: indi-qhy/qhy_focuser.h [hunk 3]
```diff
@@ -29,47 +33,57 @@ class QFocuser : public INDI::Focuser
...
+  int32_t cmd_version;
+  int32_t cmd_version_board;
   int32_t cmd_position;
   int32_t cmd_out_temp;
   int32_t cmd_chip_temp;
+  int32_t cmd_voltage;
+
 
   ISwitch ResetPosS[1];
   ISwitchVectorProperty ResetPosSP;
```
Comment (knro) [inline indi-qhy/qhy_focuser.h:71]: 1. Please use INDI::PropertySwitch
2. How is this different from focuser SYNC? Sync takes whatever value and resets the focuser to this position.

**Comment by knro** [comment]
@NYiwtisolatedall Can you please check the build errors?

**Pull request merged**